frontend/src/app/main/ui/workspace/sidebar/assets/colors.cljs

```diff
@@ -386,17 +386,26 @@
         (mf/use-fn
          (mf/deps file-id)
          (fn [event]
-           (st/emit! (dw/set-assets-section-open file-id :colors true)
-                     (ptk/event ::ev/event {::ev/name "add-asset-to-library"
-                                            :asset-type "color"}))
-           ;; FIXME: replace interop with dom helpers
-           (modal/show! :colorpicker
-                        {:x (.-clientX event)
-                         :y (.-clientY event)
-                         :on-accept add-color
-                         :data {:color "#406280"
-                                :opacity 1}
-                         :position :right})))
+           (let [x-position (-> event
+                                (dom/get-current-target)
+                                (dom/get-bounding-rect)
```

Comment:
This code is repeated for y-position and x-position. I think we can reuse the bounding rect calculation.